Homework 25

xXFranzFXx opened the pull request:
Franz Fernando A51 homework-25

xXFranzFXx committed:
initial commit

xXFranzFXx committed:
creates addSongToPlayist test case

xXFranzFXx committed:
syntax corrections for chromedriver

xXFranzFXx committed:
homework 18 initial commit

xXFranzFXx committed:
sets up @Test playSong method, configures TestNG xml

xXFranzFXx committed:
sets up helper methods for playSong test case

xXFranzFXx committed:
creates playSong testCase

xXFranzFXx committed:
initial commit

xXFranzFXx committed:
updates homework 19

xXFranzFXx committed:
creates deletPlaylist test case

xXFranzFXx committed:
initial commit

xXFranzFXx committed:
refactors homework 19 with selenium waits

xXFranzFXx committed:
refactors basetest  class

xXFranzFXx committed:
initial commit

xXFranzFXx committed:
creates renamePlaylist method

xXFranzFXx committed:
Creates BasePage Object

xXFranzFXx committed:
Creates HomePage object

xXFranzFXx committed:
creates login test using POM

xXFranzFXx committed:
Syntax corrections

xXFranzFXx committed:
refactors BasePage to SPF

xXFranzFXx committed:
refactors LoginPage to SPF

xXFranzFXx committed:
Refactors HomePage and LoginPage for SPF

xXFranzFXx committed:
reuses homework22 for SPF test case

xXFranzFXx committed:
initial commit

sets up selenium grid

xXFranzFXx committed:
syntax corrections

xXFranzFXx committed:
initial commit

sets up lambdatest

xXFranzFXx committed:
homework-25

sets up lambdatest for selenium parallel testing

Berichicko commented:
You got this problem, please correct it. Please finish your homework.

![image](https://github.com/Test-Pro-Bootcamp/A51-QA-Automation-Framework/assets/104761482/76703c8e-28ee-423f-9d72-4358ad909561)

![image](https://github.com/Test-Pro-Bootcamp/A51-QA-Automation-Framework/assets/104761482/2e7242ea-3bfd-4949-a320-47e6a15051e1)

HINT

I'll try to help you, here's an example of parallel testing in 4 threads on a local machine:



_Download_

1) Code with Russian comments https://drive.google.com/file/d/1vD6sCpCuJbBaa3JexJQJPMbL7LHQKiIG/view?usp=sharing


2) Code with English comments 
https://drive.google.com/file/d/1UT177UX39eWqt9BkxTgDFrsqG5AZ_OLU/view?usp=sharing


**To run the tests, open the README file in IntelliJ IDEA**


![image](https://github.com/Test-Pro-Bootcamp/A45-QA-Automation-Framework/assets/104761482/3c7532eb-65f0-4854-823b-be41ba43622a)

**click**  ![image](https://github.com/Test-Pro-Bootcamp/A45-QA-Automation-Framework/assets/104761482/7313f1bd-6aba-4a26-8f83-dd510a432c8c)

_PS:  To unzip a ZIP file on a Mac_ https://www.winzip.com/en/learn/file-formats/zip/mac/

**Do I have to run tests in Lambdatest or on my local machine?**
_It depends on your needs. Lambdatest is for cross-browser and device testing, but running tests locally is fine_

xXFranzFXx committed:
syntax corrections

removes unnecessary files, makes syntax corrections to BaseTest, makes syntax corrections to BasePage

Berichicko commented:
Hello,

I have checked your homework 25, and I wanted to let you know that you're doing great! Keep up the excellent work!
Best wishes from the HW team (Alex, [aliaksandr@testpro.io](mailto:aliaksandr@testpro.io))!

![804deb64dc406a20a75da05d91a1a69e](https://user-images.githubusercontent.com/104761482/235961479-9afb2841-b657-46c9-99f8-2402f933b843.jpg)

![image](https://github.com/Test-Pro-Bootcamp/A51-QA-Automation-Framework/assets/104761482/e3c5eb24-ac32-4663-a002-38ae297e493e)
